jolt-core/src/r1cs/snark.rs

```diff
@@ -73,73 +73,54 @@ impl<F: PrimeField<Repr = [u8; 32]>> Circuit<F> for JoltCircuit<F> {
 
     let cfg: CircomConfig<F> = CircomConfig::new(wtns_path.clone(), r1cs_path.clone()).unwrap();
 
-    let variable_names = [
-      "prog_a_rw", 
-      "prog_v_rw", 
-      "memreg_a_rw", 
-      "memreg_v_reads", 
-      "memreg_v_writes", 
-      "chunks_x", 
-      "chunks_y", 
-      "chunks_query", 
-      "lookup_output", 
-      "op_flags"
+    let variable_names: Vec<String> = vec![
+      "prog_a_rw".to_string(), 
+      "prog_v_rw".to_string(), 
+      "memreg_a_rw".to_string(), 
+      "memreg_v_reads".to_string(), 
+      "memreg_v_writes".to_string(), 
+      "chunks_x".to_string(), 
+      "chunks_y".to_string(), 
+      "chunks_query".to_string(), 
+      "lookup_output".to_string(), 
+      "op_flags".to_string(),
+      "input_state".to_string()
```

Comment:
Nit: Can we move input state to the top? Just feels right.

Reply:
Unfortunately non-trivial to do this due to the zip.

Could add some chaining logic but I can't make it pretty.

Reply:
Yeah I figured. I think that's why I kept it last too.